testsuite/tests/typing-layouts/illegal-across-modules/test.ml

```diff
@@ -0,0 +1,14 @@
+(* TEST
+   readonly_files = "shadow_stdlib.ml bug.ml ref.ml";
+   setup-ocamlc.byte-build-env;
+   module = "ref.ml";
+   flags = "-allow-illegal-crossing";
+   ocamlc.byte;
+   module = "shadow_stdlib.ml";
+   flags = "";
+   ocamlc.byte;
+   flags = "-allow-illegal-crossing";
+   module = "bug.ml";
+   ocamlc.byte;
+   check-ocamlc.byte-output;
+ *)
```

Comment:
I'm surprised that the `-allow-illegal-crossing` flag has anything to do with this error since illegal crossing isn't actually used anywhere. But I do agree that the error only occurs when the flag is included. Do you know what's going on here? I had thought that maybe the check in `Type_decl.check_kind_coherence` that happens only when illegal crossing is enabled was responsible, but based on testing, that doesn't seem to be the case. This leads me to the conclusion that `add_portability_and_contention_crossing` isn't a no-op when no annotation is written, which I didn't expect.

Reply:
`type_jkind` sometimes gets stuff other than an annotation. I don't think it's worth the archaeology here, especially because `-allow-illegal-crossing` is on the chopping block.